graphql_client_cli/src/generate.rs

```diff
@@ -25,8 +26,44 @@ pub fn generate_code(
         additional_derives,
         deprecation_strategy,
     };
+
     let gen = generate_module_token_stream(query_path, schema_path, Some(options))?;
-    let mut file = File::create(output)?;
-    write!(file, "{}", gen.to_string());
+
+    let mut file = File::create(output.clone())?;
+
+    let codes = gen.to_string();
+
+    if cfg!(feature = "rustfmt") && !no_formatting {
+        let codes = format(&codes);
+        write!(file, "{}", codes);
+        return Ok(());
+    }
+
+    write!(file, "{}", codes);
+
     Ok(())
```

Comment:
It's not super important, but the `write!` and `Ok(())` lines are repeated. We could write `let code = if cfg!(...) && !no_formatting { .... } else { .... };` and have the `write` and `Ok` only once.